Update list of performance improvement tips (#5085)
aframevr/aframe: docs/introduction/best-practices.md

[geometrymerger]: https://www.npmjs.com/package/aframe-geometry-merger-component

> Performance is critical in VR. People feel comfortable when you maintain a high frame rate.

Reviewer: keeping frame right high is a `must`. The re-wording makes it look less important

Author: ```suggestion
Performance is critical in VR. To make people feel comfortable, you must maintain a high frame rate.
```

Author: Or even shorter (with or without bold formatting):

```suggestion
Performance is critical in VR. You **must** maintain a high frame rate.
```

> Check out the [recommended hardware specifications][hardware]. Generally, the fewer number of entities and lights in the scene, the better.

Reviewer: This tip is a bit too general and might be even misleading in some cases.

Author: Here with "Use", I just copied the second part from line 93.  

```suggestion
Use the [recommended hardware specifications][hardware]. Generally, the fewer entities and lights in the scene, the better.
```

> Note that the **[stats component][stats]** provides various metrics (FPS, vertex and face count, geometry and material count, draw calls, number of entities). 

Reviewer: Use is a better call to action than note I think

Author: ```suggestion
Use the **[stats component][stats]** for various metrics (FPS, vertex and face count, geometry and material count, draw calls, number of entities). 
```

We want to maximize FPS and minimize everything else.

Some general tips to improve the performance of an A-Frame scene:

- Limit draw calls as much as possible. Each geometry, object, model without
  optimization is a draw call. As a rule of thumb, try to keep under 300
  maximum.
- Use [geometry-merger][geometrymerger] to merge all static meshes if possible and then make use a three.js
  material with vertex colors enabled. three.js geometries keep data (e.g. color, uvs per vertex, etc.) to maintain and manipulate geometries
  post-merge.
- Use the **[asset management system][asm]** for browser caching and
  asset preloading.
- Ensure your textures' resolutions are sized to powers of two (e.g.,
  256x256, 512x1024) to avoid the renderer having to resize the
  texture during runtime.
- Limit the number of faces and vertices on models.
- Use common 3D and game industry performance techniques (e.g., geometry
  merging, geometry instancing, level of detail, object pooling)

> - To define a solid color as the scene background and to prevent the creation of
> unnecessary geometry, use the **[background component][background]** instead of `a-sky`.

Author: Another idea here: 

```suggestion
- To define a solid color as the scene background and to prevent the creation of
unnecessary geometry, do not use `a-sky` but the **[background component][background]**.
```

  optimization is a draw call. As a rule of thumb, try to keep under 300
  maximum.
- Use [geometry-merger][geometrymerger] to merge all static meshes if possible and then make use a three.js
  material with vertex colors enabled. three.js geometries keep data (e.g. color, uvs per vertex, etc.) to maintain and manipulate geometries
  post-merge.
- Use the **[asset management system][asm]** for browser caching and
  asset preloading.
- Ensure your textures' resolutions are sized to powers of two (e.g.,
  256x256, 512x1024) to avoid the renderer having to resize the
  texture during runtime.
- Limit the number of faces and vertices on models.
- Use common 3D and game industry performance techniques (e.g., geometry
  merging, geometry instancing, level of detail, object pooling)
- To define a solid color as the scene background and to prevent the creation of
unnecessary geometry, use the **[background component][background]** instead of `a-sky`.
- To avoid overhead on `.setAttribute`, update `position`, `rotation`, `scale`, and `visible` using at the three.js
  level (`el.object3D.position`, `el.object3D.rotation`, `el.object3D.scale`,
  `el.object3D.visible`).

> - To pre-generate and reuse entities, use the **[pool component][pool]** if you need to create, remove and re-create many entities of the same type. This
>   avoids the cost of creating entities on the fly and reduces garbage collection.

Author: ```suggestion
- To pre-generate and reuse entities (if you need to create, remove and re-create many entities of the same type), use the **[pool component][pool]** to avoid creating entities on the fly and reduce garbage collection.
```

> Some tips for special techniques: 

Author: reviewer note. Unsure about this intro text. I have found the topics below which seem pretty specific to certain use cases. Maybe we need to refine the text here to `specific` or just use `Other tips`

Author: ```suggestion
More hints for models and raycasters/colliders: 
```
